conans/client/generators/__init__.py

```diff
@@ -83,10 +84,37 @@ def __contains__(self, name):
     def __getitem__(self, key):
         return self._generators[key]
 
+    @staticmethod
+    def _new_generator(generator_name):
+        if generator_name == "CMakeToolchain":
+            from conan.tools.cmake import CMakeToolchain
+            return CMakeToolchain
+        elif generator_name == "MakeToolchain":
+            from conan.tools.gnu import MakeToolchain
+            return MakeToolchain
+        elif generator_name == "MSBuildToolchain":
+            from conan.tools.microsoft import MSBuildToolchain
+            return MSBuildToolchain
+        elif generator_name == "MesonToolchain":
+            from conan.tools.meson import MesonToolchain
+            return MesonToolchain
```

Comment:
See [line 121 in this file](https://github.com/conan-io/conan/pull/8014/files#diff-e235fa62562f3e2f47330538b120750466913d3d1733d75c6eee9a06906dc95fR121), we need to add these new generators to the list, otherwise the error message won't list all of them:

```python
                raise ConanException("Invalid generator '%s'. Available types: %s" %
                                     (generator_name, ", ".join(self.available)))
```